dao.go

```diff
@@ -23,3 +23,12 @@ func (d *Dao) AddDevice(device *DevicePayload) (int, error) {
 	d.data[dev.Id] = *dev
 	return dev.Id, nil
 }
+
+func (d *Dao) GetDevice(id int) (*Device, error) {
+	for _, device := range d.data {
```

Comment:
Since this DAO uses a map you don't have to look through the whole map - just get device by its id like so
```suggestion
	if device, ok := d.data[id]; ok {
```

Reply:
no tak :)